router/src/infer/chat_template.rs

```diff
@@ -74,14 +74,38 @@ impl ChatTemplate {
                     format!("\n---\n{}", tool_prompt)
                 };
                 if let Some(last_message) = messages.last_mut() {
-                    last_message.content.push(MessageChunk::Text { text });
+                    if let Some(ref mut content) = last_message.content {
+                        content.push(MessageChunk::Text { text });
+                    } else {
+                        last_message.content = Some(MessageContent::SingleText(text));
+                    }
                 }
                 Some(tools)
             }
             None => None,
         };
 
-        let messages: Vec<TextMessage> = messages.into_iter().map(|c| c.into()).collect();
+        let messages: Vec<TextMessage> = messages
```

Comment:
Is this PR related to https://github.com/huggingface/text-generation-inference/pull/3036 ?

Maybe we should fuse them since they seem close.

In general, the error should happen directly at the parsing level, not here (so instead of 2 Option, we need a Enum, and a clear error message).

Reply:
Yes, we can close this one in favor of #3036